backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -612,31 +626,18 @@ private SessionFrame updateMatchingFrame(String frameId) {
         return null;
     }
 
-    private boolean pushNewFrame(SessionFrame matchingFrame, boolean newFrame, boolean currentFramePushed) {
-        // ok, frame should be appropriately modified now.
-        // we also need to push this frame if it's new
-        if (newFrame) {
-            // Before we can push a frame onto the stack, we need to
-            // make sure the stack is clean. This means that if the
-            // current frame has a snapshot, we've gotta make sure
-            // the existing frames are still valid.
-
-            // TODO: We might want to handle this differently in the future,
-            // so that we can account for the invalidated frames in the ui
-            // somehow.
-            cleanStack();
-
-            // OK, now we want to take the current frame and put it up on the frame stack unless
-            // this frame is dead (IE: We're closing it out). then we'll push the new frame
-            // on top of it.
-            if (!frame.isDead() && !currentFramePushed) {
-                frameStack.push(frame);
-                return true;
-            }
```

Comment:
@ctsims I removed this code, which pushes the current frame onto the top of the stack frame because it didn't make sense to me and was complicating my implementation of this extension.  (note that when I refactored this code in https://github.com/dimagi/commcare-core/pull/367 I made a mistake, the `return true` should really be `currentFramePushed = true` to persist the pre-refactor behavior)
Can you verify that this isn't needed? If it is needed, can you explain why?